lightning/src/ln/channelmanager.rs

```diff
@@ -2304,15 +2380,27 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 											let mut total_value = 0;
 											let htlcs = channel_state.claimable_htlcs.entry(payment_hash)
 												.or_insert(Vec::new());
+											if htlcs.len() == 1 {
+												if let OnionPayload::Spontaneous(_) = htlcs[0].onion_payload {
+													log_trace!(self.logger, "Failing new HTLC with payment_hash {} as we already had an existing keysend HTLC with the same payment hash", log_bytes!(payment_hash.0));
```

Comment:
Well I've a doubt if this doesn't introduce a weird payment censorship attack. Let's say an attacker observe that a payment invoice hash X has been issued by Alice. From now on, it will repeatedly send keysend X to Alice aiming to provoke a collision with honest HTLC X, thus triggering a reject.

Is the concern exposed sound ? I don't think the attacker even has to be on the honest HTLC payment path...

Reply:
Yea, but its solved by https://github.com/rust-bitcoin/rust-lightning/pull/967#discussion_r670539027 as well, no?